### PR TITLE
[JBPM-9436]  AMQ Streams (Kafka) listener functionality / consumer

### PR DESCRIPTION
**Pull request by fjtirado**
Adding Kafka dependency to server common so it is added to j2ee war and spring boot starters

**JIRA**: 

[JBPM-9436](https://issues.redhat.com/browse/JBPM-9436)

Depends on: 
[#1495](https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/1495)

### CODE REVIEW

File: drools-ha/pom.xml [hunk 1]
```diff
@@ -24,7 +24,7 @@
     <surefire.forkCount>1</surefire.forkCount>
     <version.javax.el>3.0.1-b08</version.javax.el>
     <version.javax.ws.rs>2.0.0.Final</version.javax.ws.rs>
-    <version.org.apache.kafka>2.2.1</version.org.apache.kafka>
+    <version.org.apache.kafka_2>2.2.1</version.org.apache.kafka_2>
```
Comment (elguardian) [inline drools-ha/pom.xml:27]: why is this change required ?

**Comment by elguardian** [comment]
I am going to reject this change. You are packaging a dependency that is not needed for this JIRA. So there is no point to packaging in kie server when the event emitter is something optional.

**Comment by fjtirado** [comment]
> I am going to reject this change. You are packaging a dependency that is not needed for this JIRA. So there is no point to packaging in kie server when the event emitter is something optional.
> 
> I might be the case that for a another jira (listener) this could be needed. but let's not mix stuff.

The version of kafka client to be use on jbpm-workitem should match the one used in emitter, thats the reason for this PR. The version for kafka should be on kie parent

File: drools-ha/pom.xml (hunk 1 above)
Comment (fjtirado) [reply drools-ha/pom.xml:27]: This is a typo
